CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java

```diff
@@ -31,24 +60,96 @@ public ProfileFragment() {
      * @return A new instance of fragment ProfileFragment.
      */
     // TODO: Rename and change types and number of parameters
-    public static ProfileFragment newInstance(String param1, String param2) {
+    public static ProfileFragment newInstance(String username) {
         ProfileFragment fragment = new ProfileFragment();
         Bundle args = new Bundle();
+        args.putString("username", username);
         fragment.setArguments(args);
         return fragment;
     }
 
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        thisContext = context;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        FirebaseFirestore db;
+        db = getFirestoreInstance();
+
+        confirmUserbtn.setOnClickListener(view1 -> {
+            Map<String, Object> user = new HashMap<>();
```

Comment:
This function is very overloaded. You can add a new user, edit a user, and switch to another user using the same button. We can keep it like this for now, but the functionalities will need to be split in the future.